### ISSUE TITLE
clojurescript variants that have nodejs pre-installed?

### ISSUE THREAD

**Issue by cap10morgan** [Quantisan/docker-clojure#91]
Should we consider adding clojurescript images that pre-install node, npm, etc.?

**Comment by njordhov** [comment]
Yes! I am currently using a custom Dockerfile for ClojureScript development that builds on the `Clojure` image to pre-install nodejs.

As a first step, I suggest making `clojure:default` also have nodejs pre-installed, providing a kitchen-sink image for clojure(script) development.

**Comment by cap10morgan** [comment]
@njordhov Hmm... interesting idea. I was thinking of just creating official `clojurescript` images with all the other tags we generate but with latest LTS node / npm stuff pre-installed and the latest clojurescript pre-downloaded. And then `clojurescript:latest` would have lein, boot, and tools-deps stuff pre-installed like we recently started doing with `clojure:latest`. What do you think of that?

Thoughts @Quantisan?

**Comment by Quantisan** [comment]
@cap10morgan a separate image sounds good to keep it clean. Otherwise, I haven't used clojurescript for ages so I don't know what toolsets are needed anymore.

**Comment by njordhov** [comment]
@cap10morgan A separate clojurescript image is fine, particularly when going beyond a kitchen-sink image. Thanks for all your work in making this happen.

**Comment by souenzzo** [comment]
Please no; AFIK there is no cljs builder that uses node in process.

You just need the node_modules folder.

Just do like this:
```Dockerfile
FROM node:alpine AS node
COPY package.json package-lock.json ./
RUN npm install

FROM openjdk-16-tools-deps-alpine
COPY --from=node node_modules node_modules
RUN clojure -A:shadow-cljs release app 
```

**Comment by cap10morgan** [comment]
@souenzzo That's an argument for why you don't need it (not why it would be undesirable nor why others shouldn't need it). If we created it and you still didn't need it, you could just not use it. The other images aren't going away.

**Comment by call-a3** [comment]
@souenzzo I still have a use-case for this, using leiningen/fighweel-main which uses webpack for post-processing/bundling.

**Comment by cap10morgan** [comment]
This is still something I'd like to do. I think it will involve changing some fundamental assumptions in our build system, so the work involved there is the big hurdle.

**Comment by lread** [comment]
Tangental tip for those using the clojure docker image with CircleCI and wanting to add nodejs support: you can add nodejs easily enough with their [node orb](https://circleci.com/developer/orbs/orb/circleci/node). Here's how [I did this for cljdoc's build](https://github.com/cljdoc/cljdoc/commit/6b0c160764956683a911898d421eab908f8efd93).
